statsd/client_test.go

```diff
@@ -54,36 +53,32 @@ func TestClientUDP(t *testing.T) {
 	}
 	defer server.Close()
 
-	client, err := statsd.New(addr)
+	client, err := New(addr)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	clientTest(t, server, client)
 }
 
-type statsdWriterWrapper struct {
-	io.WriteCloser
-}
-
 func TestClientWithConn(t *testing.T) {
 	server, conn, err := os.Pipe()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	client, err := statsd.NewWithWriter(statsdWriterWrapper{conn})
+	client, err := NewWithWriter(conn)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	clientTest(t, server, client)
 }
 
-func clientTest(t *testing.T, server io.Reader, client *statsd.Client) {
+func clientTest(t *testing.T, server io.Reader, client *Client) {
 	for _, tt := range dogstatsdTests {
-		client.Namespace = tt.GlobalNamespace
-		client.Tags = tt.GlobalTags
+		client.namespace = tt.GlobalNamespace
+		client.tags = tt.GlobalTags
```

Comment:
I know these are tests but I'm not sure that we should be trying to use non-public fields here otherwise we're not really testing the same thing that the user will be interacting with.Is there a way to run these with the public side of the APIs?

Reply:
Very good point. The tests are a bit of a mess right now. It's one of the main improvement I want to bring in V5. We already included some test helpers that will ease the testing issue and will allow us to rewrite those test in a better way (and remove duplicated tests).

It will be done is a future PRs